### ISSUE TITLE
Rename/replace `AnnexRepo.file_has_content()`

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2055]
I have been tripping over this function so often that it reached my threshold to severely dislike it. I always assume that this function tests whether a "file has content" -- which is not what it does. It checks whether a file is symlinked into an annex. It returns False for any file in git and also returns false for files (with content) in direct mode. This function is rarely used outside the tests:

```
% git grep '\.file_has_content' |grep -v test
datalad/auto.py:        if (under_annex or under_annex is None) and not annex.file_has_content(filepath):
datalad/crawler/nodes/annex.py:        #         if self.repo.file_has_content(fpath):
datalad/interface/add_archive_content.py:                # if annex.file_has_content(target_path):
datalad/interface/ls.py:                    if self.repo.file_has_content(self._path) \
datalad/interface/unlock.py:                    ds.repo.file_has_content(files)):
datalad/metadata/metadata.py:        content_info = zip(paths, ds.repo.file_has_content(paths))
datalad/plugin/export_archive.py:            has_content = repo.file_has_content(
datalad/support/annexrepo.py:            return [f for f in files if self.file_has_content(f)]
```

Use of this function needs to be wrapped in complex additional tests, when not done it is likely a bug (e.g. quick glance makes me think use in `export_archive` is wrong for direct mode).

In summary, we don't need what this function does. if it is needed in the tests (and it is heavily used there) it should become a test helper.

I propose to have a function called `AnnexRepo.has_content()` that returns True whenever the file has content present -- regardless of its mood.

Demo of my struggle:

```
(Pdb) fname
u'/tmp/datalad_temp_test_niftiDP0rbx/nifti1.nii.gz'
(Pdb) import nibabel as nb
(Pdb) nb.load(fname).shape
(91, 109, 91, 2)
(Pdb) ds.repo.file_has_content(fname)
False
(Pdb) ds.repo.file_has_content(fname, allow_quick=False)
False
(Pdb) ds.repo
<AnnexRepo path=/tmp/datalad_temp_test_niftiDP0rbx (<class 'datalad.support.annexrepo.AnnexRepo'>)>
(Pdb) ds.repo.is_under_annex(fname)
False
```

**Comment by yarikoptic** [comment]
May be it is just a bug and it was intended to check for the content originally?

**Comment by mih** [comment]
Closing in favor of #1100
